### PR TITLE
Improve errors handling

### LINKED ISSUE #5: Client's validateResponse() method is not handling 401 HTTP correctly
When client send `getUser()` request, and receive response with HTTP 401 code and message Unauthorized, the `validateResponse()` method does not catch this correctly.

Error was introduced by sending token value as `Bearer Bearer xyz` instead of `Bearer xyz`.

Identifier service response content for getUser() endpoint and invalid token:
```
{
  "errors": [
    {
      "code": "token",
      "message": "invalid authorization token",
      "params": []
    }
  ]
}
```

Client must handle response 401 with specific code `token`.

### LINKED ISSUE #7: Improve errors handling
Sometimes Client cannot handle invalid response form Identifier service then throw unexpected PHP errors without proper information.

```
Insly\Identifier\Client\Entities\Builders\UserBuilder::buildFromResponse(): Argument #1 ($userData) must be of type array, null given, called in /var/www/build/vendor/insly/identifier-service-phpclient/src/Client.php on line 156
```

Client should handle all invalid responses from Identifier service and attach error messages from Identifier service also to the response.

### PR DESCRIPTION
**Pull request by mtracz**
In this PR I refactored handling error responses. From now an exception should be thrown if response status is not 200.
Improved setting up host in `Config()`. From now, it doesn't matter if host ends with `/` or not. Both cases are valid.
Improved `buildHeaders()` method, cause it was not working with PHP 8.0 as expected.
Added tests;
Added xDebug;
Added Makefile with helpful commands;
Updated Composer packages
Updated readme;

Closes #5 
Closes #7

### CODE REVIEW

File: readme.md [hunk 1]
```diff
@@ -77,7 +77,10 @@ class CognitoServiceProvider extends ServiceProvider
     {
         $this->app->bind(Client::class, function (): Client {
             $config = new Config("https://example.com/api/v1/", "tenant");
-            $config->setToken(config("auth.cognito.token"));
+            
+            $authorizationHeader = $request?->header("Authorization") ?? "";
```
Comment (krzysztofrewak) [inline readme.md:81]: There is no $request here.

File: composer.json [hunk 2]
```diff
@@ -11,13 +11,13 @@
...
   "require": {
     "php": "^8.0",
-    "guzzlehttp/guzzle": "^7.2",
-    "symfony/http-foundation": "^5.3"
+    "guzzlehttp/guzzle": "^7.4",
+    "symfony/http-foundation": "^5.4"
   },
   "require-dev": {
-    "blumilksoftware/codestyle": "^0.4",
+    "blumilksoftware/codestyle": "^0.7",
```
Comment (krzysztofrewak) [inline composer.json:18]: Bump it to 0.8.0

File: tests/mocked/ErrorResponseHandlerTest.php [hunk 3]
```diff
@@ -0,0 +1,314 @@
...
+        $responseArray = Standard::getUndefinedErrorCodeExceptionResponse();
+        $response = new Response(status: 400, body: json_encode($responseArray));
+
+        $client = new Client(new GuzzleClient(), new Config("host", "tenant"));
+        $validateResponseMethod = $this->getProtectedMethod(Client::class, "validateResponse");
+
+        // then
+        $this->expectException(UndefinedErrorCodeException::class);
+
+        // when
```
Comment (krzysztofrewak) [inline tests/mocked/ErrorResponseHandlerTest.php:91]: Shouldn't it be given-when-then?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: tests/mocked/ErrorResponseHandlerTest.php (hunk 3 above)
Comment (mtracz) [reply tests/mocked/ErrorResponseHandlerTest.php:91]: Theoretically, but we cannot move line 89 (expect Exception) after line where the exception is thrown. It won't work.

File: readme.md (hunk 1 above)
Comment (mtracz) [reply readme.md:82]: Updated readme with 
```php
$request = request();
```

File: composer.json (hunk 2 above)
Comment (mtracz) [reply composer.json]: Bumped

File: readme.md (hunk 1 above)
Comment (krzysztofrewak) [reply readme.md:82]: I'd prefer something like this:
```php
        $this->app->bind(Client::class, function (Application $app): Client {
             $request = $app->make(Request::class);
             $authorizationHeader = $request?->header("Authorization") ?? ""; 
```

File: tests/mocked/ErrorResponseHandlerTest.php (hunk 3 above)
Comment (krzysztofrewak) [reply tests/mocked/ErrorResponseHandlerTest.php:91]: Then I think it's really misguiding here. I would remove these comments.

File: docker/php/Dockerfile [hunk 4]
```diff
@@ -0,0 +1,19 @@
...
+
+RUN if [ ${INSTALL_XDEBUG} = true ]; then \
+    apk --no-cache add $PHPIZE_DEPS \
+    && pecl install xdebug-${XDEBUG_VERSION} \
+    && docker-php-ext-enable xdebug \
+    && echo "xdebug.client_host=${XDEBUG_CLIENT_HOST}" >> /usr/local/etc/php/conf.d/xdebug.ini \
+    && echo "xdebug.client_port=${XDEBUG_CLIENT_PORT}" >> /usr/local/etc/php/conf.d/xdebug.ini \
+    && echo "xdebug.mode=${XDEBUG_MODE}" >> /usr/local/etc/php/conf.d/xdebug.ini \
+    && echo "xdebug.start_with_request=${XDEBUG_START_WITH_REQUEST}" >> /usr/local/etc/php/conf.d/xdebug.ini \
+;fi
```
Comment (krzysztofrewak) [inline docker/php/Dockerfile:19]:
```suggestion
;fi

```

File: src/Config.php [hunk 5]
```diff
@@ -28,15 +23,10 @@ public function getHost(): string
         return $this->host;
     }
 
-    /**
-     * @throws InvalidHostFormatException
-     */
-    public function setHost(string $host, bool $validate = true): void
+    public function setHost(string $host): void
```
Comment (krzysztofrewak) [inline src/Config.php:26]: You are changing public method in one of most important classes in the project. Maybe you could leave this optional `$validate` and mark it deprecated and remove in next version?

File: src/DTO/IdentifierService/IdentifierErrorResponse.php [hunk 6]
```diff
@@ -0,0 +1,43 @@
...
+declare(strict_types=1);
+
+namespace Insly\Identifier\Client\DTO\IdentifierService;
+
+use Insly\Identifier\Client\DTO\IdentifierService\Response\Error;
+
+class IdentifierErrorResponse
+{
+    /**
+     * @param Error[] $errors
```
Comment (krzysztofrewak) [inline src/DTO/IdentifierService/IdentifierErrorResponse.php:12]:
```suggestion
     * @param array<Error> $errors
```

We are thinking about implementing some rules from [here](https://github.com/kubawerlos/php-cs-fixer-custom-fixers) into our codestyle package. Then we'd force to use generic array style in PHPDocs, but I think we can start doing it right now.

File: src/DTO/IdentifierService/IdentifierErrorResponse.php [hunk 7]
```diff
@@ -0,0 +1,43 @@
...
+     */
+    public function __construct(
+        protected array $errors,
+        protected int $statusCode,
+    ) {
+    }
+
+    public static function fromResponseContent(array $responseContent, int $statusCode): static
```
Comment (krzysztofrewak) [inline src/DTO/IdentifierService/IdentifierErrorResponse.php:20]: Why won't you create a builder class for that?

**Review by krzysztofrewak — CHANGES_REQUESTED** [review]

File: tests/mocked/ErrorResponseHandlerTest.php (hunk 3 above)
Comment (mtracz) [reply tests/mocked/ErrorResponseHandlerTest.php]: Removed comments. Moved expectException to the top

File: readme.md (hunk 1 above)
Comment (mtracz) [reply readme.md:84]: Changed

File: src/DTO/IdentifierService/IdentifierErrorResponse.php (hunk 7 above)
Comment (mtracz) [reply src/DTO/IdentifierService/IdentifierErrorResponse.php]: Added builder class

**Review by krzysztofrewak — APPROVED** [review]

**Pull request merged**